dist/common/scripts/scylla_util.py

```diff
@@ -117,6 +117,239 @@ def curl(url, headers=None, byte=False, timeout=3, max_retries=5, retry_interval
                 raise
 
 
+class azure_instance:
+    """Describe several aspects of the current AZURE instance"""
+
+    EPHEMERAL = "ephemeral"
+    ROOT = "root"
+    GETTING_STARTED_URL = "http://www.scylladb.com/doc/getting-started-azure/"
+    META_DATA_BASE_URL = "http://169.254.169.254/metadata/instance"
+    ENDPOINT_SNITCH = "SimpleSnitch"
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+        self.__firstNvmeSize = None
+        self.__osDisks = None
+
+    @staticmethod
+    def is_azure_instance():
+        """Check if it's AZURE instance via DNS lookup to metadata server."""
+        import socket
+        try:
+            addrlist = socket.getaddrinfo('169.254.169.254', 80)
+        except socket.gaierror:
+            return False
+        for res in addrlist:
+            af, socktype, proto, canonname, sa = res
+            if af == socket.AF_INET:
+                addr, port = sa
+                if addr == "169.254.169.254":
+                    return True
```

Comment:
Can you point me to the documentation that describes how to detect Azure? IOW, what you used as a source to implement this check.